lib/spack/spack/database.py

```diff
@@ -393,7 +393,7 @@ def _add(self, spec, path, directory_layout=None):
 
         else:
             self._data[key] = InstallRecord(spec, path, True)
-            for dep in spec.dependencies.values():
+            for dep in spec.dependencies(spack.DEP_INSTALL):
```

Comment:
The ref count is intended to keep around the record in the DB until nothing refers to the missing spec.  If you set the ref count only based on *install* dependencies, then you can still run a `spack find -d` and get errors, because remaining `spec.yaml` files may refer to a build or run dependencies that have since been uninstalled. You want to keep the record in the DB to handle cases like this -- the DB is supposed to tell us not only what is installed but also what is missing, and it keeps refs around until nothing could refer to the spec anymore.

Aside: Even if the ref count were just used to track what should be uninstalled, this would be close, but I think you would want to track both install and run dependencies here, as you don't want to allow something to be uninstalled if it's still a run dependency of something else.